contracts/bridge/contracts/common/fee-market/FeeMarket.sol

```diff
@@ -0,0 +1,448 @@
+pragma solidity >=0.6.0 <0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "../../interfaces/IFeeMarket.sol";
+
+contract FeeMarket is IFeeMarket {
+    event SetOwner(address indexed o, address indexed n);
+    event SetOutbound(address indexed out, uint256 flag);
+    event SetParaTime(uint32 slash_time, uint32 relay_time);
+    event SetParaRelay(uint32 assigned_relayers_number, uint256 collateral);
+    event Slash(address indexed src, uint wad);
+    event Reward(address indexed dst, uint wad);
+    event Deposit(address indexed dst, uint wad);
+    event Withdrawal(address indexed src, uint wad);
+    event Locked(address indexed src, uint wad);
+    event UnLocked(address indexed src, uint wad);
+    event AddRelayer(address indexed prev, address indexed cur, uint fee);
+    event RemoveRelayer(address indexed prev, address indexed cur);
+    event OrderAssgigned(uint256 indexed key, uint256 timestamp, uint32 assigned_relayers_number, uint256 collateral);
+    event OrderSettled(uint256 indexed key, uint timestamp);
+
+    address private constant SENTINEL_HEAD = address(0x1);
+    address private constant SENTINEL_TAIL = address(0x2);
+    // System treasury
+    address public immutable VAULT;
+
+    // SlashAmount = CollateralPerorder * LateTime / SlashTime
+    uint32 public SlashTime;
+    // Time assigned relayer to relay messages
+    uint32 public RelayTime;
+    // Fee market assigned relayers numbers
+    uint32 public AssignedRelayersNumber;
+    // The collateral relayer need to lock for each order.
+    uint256 public CollateralPerorder;
+    // Governance role to decide which outbounds message to relay
+    address public owner;
+    // Outbounds which message will be relayed by relayers
+    mapping(address => uint256) public outbounds;
+
+    // Balance of the relayer including deposit and eared fee
+    mapping(address => uint256) public balanceOf;
+    // Locked balance of relayer for relay messages
+    mapping(address => uint256) public lockedOf;
+    // All relayers in fee-market, they are linked one by one and sorted by the relayer fee asc
+    mapping(address => address) public relayers;
+    // relayer count
+    uint256 public relayer_count;
+    // Maker fee of the relayer
+    mapping(address => uint256) public feeOf;
+
+    struct Order {
+        // Assigned time of the order
+        uint32 assigned_time;
+        // Assigned number of relayers
+        uint32 assigned_relayers_number;
+        // Assigned collateral of each relayer
+        uint256 collateral;
+    }
+    // message_encoded_key => Order
+    mapping(uint256 => Order) public orderOf;
+    // message_encoded_key => assigned_slot => assigned_relayer
+    mapping(uint256 => mapping(uint256 => address)) public assigned_relayers;
```

Comment:
https://docs.soliditylang.org/en/v0.5.3/style-guide.html#naming-conventions

Not quite sure about naming conventions, but it's a good time to do clean up now.

Reply:
There I use `snake_case` for `order` is a good distinction for general storage variables

Reply:
For function.
`snake_case` function is  for relayer to call.
`camelCase` function is for all user to call.